bundles/org.openhab.automation.jythonscripting/src/main/java/org/openhab/automation/jythonscripting/internal/JythonScriptEngineFactory.java

```diff
@@ -36,93 +37,84 @@
  *
  * @author Scott Rushworth - Initial contribution
  * @author Wouter Born - Initial contribution
+ * @author Holger Hees - Further development
  */
 @Component(service = ScriptEngineFactory.class)
 @NonNullByDefault
 public class JythonScriptEngineFactory extends AbstractScriptEngineFactory {
 
-    private static final String PYTHON_CACHEDIR = "python.cachedir";
     private static final String PYTHON_HOME = "python.home";
+    private static final String PYTHON_HOME_PATH = JythonScriptEngineFactory.class.getProtectionDomain().getCodeSource()
+            .getLocation().toString().replace("file:", "");
+
     private static final String PYTHON_PATH = "python.path";
+    private static final String PYTHON_DEFAULT_PATH = Paths
+            .get(OpenHAB.getConfigFolder(), "automation", "jython", "lib").toString();
+
+    private static final String PYTHON_CACHEDIR = "python.cachedir";
+    private static final String PYTHON_CACHEDIR_PATH = Paths
+            .get(OpenHAB.getUserDataFolder(), "cache", JythonScriptEngineFactory.class.getPackageName(), "cachedir")
+            .toString();
 
-    private static final String DEFAULT_PYTHON_PATH = Paths
-            .get(OpenHAB.getConfigFolder(), "automation", "lib", "python").toString();
+    private static final org.python.jsr223.PyScriptEngineFactory factory = new org.python.jsr223.PyScriptEngineFactory();
 
-    private static final String SCRIPT_TYPE = "py";
-    private static final javax.script.ScriptEngineManager ENGINE_MANAGER = new javax.script.ScriptEngineManager();
+    private final List<String> scriptTypes = (List<String>) Stream.of(factory.getExtensions(), factory.getMimeTypes())
+            .flatMap(List::stream) //
+            .collect(Collectors.toUnmodifiableList());
```

Comment:
In newer Java versions you can use:

```suggestion
            .toList();
```